### ISSUE TITLE
Confusing behaviour using Huey in my Django project

### ISSUE THREAD

**Issue by ovalseven8** [coleifer/huey#342]
After multiple hours trying to debug this, I am really lost and would like to ask for help.

First of all, I am using `huey` in my Django project and started the process of integrating it.
So, I added `huey.contrib.djhuey` to my `INSTALLED_APPS` in the `settings.py` file. I've not added anything more, so it should use the default settings. Because `DEBUG = True` in the settings file, `always_eager` should also be set to True.

So, my first confusion is why I get the following output when running `python manage.py run_huey` (redis server is not started yet):
```
python manage.py run_huey

redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused.
```

Because, as far as I understand the docs, Huey should not use `redis` at all when `always_eager` is set to True.

Then, when I start the `redis` server, I get this output by the way:
```
python manage.py run_huey

[2018-08-18 23:16:00,136] INFO:huey.consumer:MainThread:Huey consumer started with 1 thread, PID 2709
[2018-08-18 23:16:00,136] INFO:huey.consumer:MainThread:Scheduler runs every 1 second(s).
[2018-08-18 23:16:00,136] INFO:huey.consumer:MainThread:Periodic tasks are enabled.
[2018-08-18 23:16:00,136] INFO:huey.consumer:MainThread:UTC is enabled.
[2018-08-18 23:16:00,136] INFO:huey.consumer:MainThread:The following commands are available:
...
```

---

Okay, but now I restructured my Django settings and created multiple settings files instead of the default one.
That means, instead of the `settings.py` file I now have a folder `settings` that contains an empty `__init__.py`, `base.py` and `development.py`.

To make it work, I've of course updated my `manage.py` file as well. Now, it looks like this:
```
#!/usr/bin/env python
import sys

import os

if __name__ == "__main__":
    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "my_app.settings.development")
    try:
        from django.core.management import execute_from_command_line
    except ImportError as exc:
        raise ImportError(
            "Couldn't import Django. Are you sure it's installed and "
            "available on your PYTHONPATH environment variable? Did you "
            "forget to activate a virtual environment?"
        ) from exc
    execute_from_command_line(sys.argv)
```

Note here that the only change is that it's now `my_app.settings.development` instead of `my_app.settings`. Okay, so everything should work.

But, well, now `huey` does not output anything when running `python manage.py run_huey`.
```
python manage.py run_huey


```

My `development.py` file is only those few lines:
```
from .base import *

DEBUG = True

INSTALLED_APPS += [
    'debug_toolbar',
]

MIDDLEWARE += ['debug_toolbar.middleware.DebugToolbarMiddleware', ]
```

So, out of debugging reasons I've changed it to only include that now:
```
from .base import *

DEBUG = True
```

It still does not work, now let's change it to `DEBUG = False`:
```
from .base import *

DEBUG = False
```

And ... now I get output again like I described above. That means in case `redis` is running:
```
python manage.py run_huey

[2018-08-18 23:29:32,489] INFO:huey.consumer:MainThread:Huey consumer started with 1 thread, PID 2964
[2018-08-18 23:29:32,489] INFO:huey.consumer:MainThread:Scheduler runs every 1 second(s).
[2018-08-18 23:29:32,489] INFO:huey.consumer:MainThread:Periodic tasks are enabled.
[2018-08-18 23:29:32,489] INFO:huey.consumer:MainThread:UTC is enabled.
[2018-08-18 23:29:32,489] INFO:huey.consumer:MainThread:The following commands are available:
...
```

That means, the change from `DEBUG = True` to `DEBUG = False` made an effect. I do no know why, however. It's not the end though.

Because when I add `django-debug-toolbar` again, it does not work (meaning I don't get any output running `python manage.py run_huey`). Even with `DEBUG = False`.
```
from .base import *

DEBUG = False

INSTALLED_APPS += [
    'debug_toolbar',
]

MIDDLEWARE += ['debug_toolbar.middleware.DebugToolbarMiddleware', ]
```

I'm really confused now. Somehow it has something to do with both if `DEBUG` is set to `True` or `False` and with `django-debug-toolbar`. Or, I just don't know what's wrong.

So, my questions/problems I do not understand:

1. Why does `huey` connect to redis when `always_eager` is set to True in my first example?
1. Because: In my example below, I get only any output when `DEBUG = False`?
1. And, adding `django-debug-toolbar` makes it not working again?

It's a long post now, I hope someone can help me, I'm kind of lost. :confused: 

### System/Packages
Python 3.7
Django 2.1
Huey 1.10.2
Redis 2.10.6
Django Debug Toolbar 1.9.1

**Comment by coleifer** [comment]
I'm not reading the whole issue because the thing that jumped out at me is... if you've got DEBUG set to `True` in your settings file, why are you trying to run the huey consumer? There's no reason to do that!

**Comment by coleifer** [comment]
Of course when you try to run the consumer it opens a connection to Redis to listen for incoming messages. When DEBUG=True (and thus huey is in "always eager" mode) there's no reason to run the consumer, though. Tasks are executed synchronously. Always-eager means the tasks are run just as if they were not decorated / normal functions.

I have no idea what "debug toolbar" may or may not be doing...jacking with log handlers? Who knows?

Huey works. I think you do not understand it, though. Closing as I do not believe this is an issue.

**Comment by ovalseven8** [comment]
Thanks for your answer.

While trying to debug this I just thought that it could have something to do with `always_eager` set to True/False. I mean, somehow it had an effect. Turns out, it also has something to do with the `urlpatterns` defined in `urls.py`.

Today I did some further tests and now I can at least isolate the problem and it's indeed a problem caused entirely by `django-debug-toolbar`.

So, you're right with "I think you do not understand it". But, honestly, this behaviour is not that easy to understand. :smile:
